src/dashboard/Dashboard.spec.js

```diff
@@ -1 +1,46 @@
-// Test away
+import React from 'react';
+import { render, fireEvent, cleanup } from 'react-testing-library';
+import renderer from 'react-test-renderer';
+
+import Dashboard from './Dashboard';
+
+beforeEach(cleanup);
+
+describe('<Dashboard />', () => {
+  it('matches snapshot', () => {
+    const tree = renderer.create(<Dashboard />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  describe('Lock/Unlock Button', () => {
+    it('does not lock when gate is open', () => {
+      const { getByText } = render(<Dashboard />);
+      fireEvent.click(getByText(/^lock gate$/i));
+      getByText(/^unlocked$/i);
+    });
+
+    it('locks when gate is closed', () => {
+      const { getByText } = render(<Dashboard />);
+      fireEvent.click(getByText(/^close gate$/i));
+      fireEvent.click(getByText(/^lock gate$/i));
+      getByText(/^locked$/i);
+    });
+  });
+
+  describe('Close/Open Button', () => {
+    it('does not open when gate is locked', () => {
+      const { getByText } = render(<Dashboard />);
+      fireEvent.click(getByText(/^close gate$/i));
+      fireEvent.click(getByText(/^lock gate$/i));
+      fireEvent.click(getByText(/^open gate$/i));
+      getByText(/^closed$/i);
```

Comment:
perfect multi-firing test case! you could also test this by explicitly passing in `closed={true} locked={true}` into a rendered  instance of the Controls, and test after only one attempted click on the `open gate` button, an argument could be made for this test to run against either component